src/atoms/firebaseUserAtoms.ts

```diff
@@ -43,6 +43,34 @@ firebaseUserAtom.onMount = setAtom => {
   };
 };
 
+// Initially it is just a dummy object, but before using it, error.credential are updated into it
+const CredentialforDataOverrideAtom = atom<{
+  providerId: string,
+  signInMethod: string,
+  oauthIdToken: string,
+  oauthAccessToken: string,
+  toJSON(): Object
+}>(
+  {
+    providerId: "google.com",
+    signInMethod: "google.com",
+    oauthIdToken: "absolutely-dummy-string",
+    oauthAccessToken: "absolutely-dummy-string",
+    toJSON: () => JSON.stringify(this)
+}
+);
+const showConfirmModal = atom<boolean>(false);
+export const respondConfirmOverrideAtom = atom(
+  get => get(showConfirmModal),
+  (get, set, dataOverrideResponse: boolean) => {
+    if(dataOverrideResponse) {
+      // get(CredentialforDataOverrideAtom) is already been updated.
+      firebase.auth().signInWithCredential(get(CredentialforDataOverrideAtom));
+    }
+    set(showConfirmModal, false);
+  }
+)
```

Comment:
These two should be combined into one atom. Also, the logic of signing in with credential should go in the `signInWithGoogle` atom, not in this atom. The show confirm modal atom should just accept a callback function; the logic for signing in should be in this callback function that's set by the `signInWithGoogle` atom